### PR TITLE
Upgrade & Convert to TS

### PR DESCRIPTION
**Pull request by JacobMGEvans**
We had many templates that were in pretty old states for TypeScript. This goes through a majority of them and updates their TSConfigs to be inline with most up to date TS configurations and Wrangler.

Additionally, many templates were written in TypeScript and had many errors in their tests and a few in their workers code (caused by updates to Wrangler & the ecosystem) converting them to TypeScript helps to fix them and helps future maintaining when updating or making changes to the templates.

Part of #3441 

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
I am reduce the scope of this, by remove adding `templates` to workspaces, that really made this look bigger than it needs to be right now.

File: templates/worker-turso-ts/package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
 		"itty-router": "^3.0.12"
 	},
 	"devDependencies": {
-		"@cloudflare/workers-types": "^4.20230321.0",
+		"@cloudflare/workers-types": "^3.0.0",
```
[the remark this answers is not in the document]
Comment (JacobMGEvans) [reply templates/worker-turso-ts/package.json:14]: Yeah seemed like they were all on v3.0.0, I think I am actually gonna do the opposite not that I have had some sleep lmao

**Review by admah — COMMENTED** [review]
Added comments on creating projects with templates. I have a PR in progress with similar instructions in each template's subdirectory.

File: templates/README.md [hunk 2]
```diff
@@ -10,29 +10,15 @@ Questions about Workers? Please join the [Cloudflare Developers Discord](https:/
...
 
-<!-- todo(eidam): is workers.new/<name> still hardcoded or ready for all? -->
-<!-- 2. In-browser development, via `workers.new` -->
-
 1. **Local development, via CLI quickstart utility**
 
-   You may use `yarn`, `pnpm`, or `npm` to invoke the [`create-cloudflare`](https://www.npmjs.com/package/create-cloudflare) package.
-
-   > **Note:** All recent versions of `npm`, `yarn`, and `pnpm` support this feature!
+   We recommend using the `npx create cloudflare@1 [folder-name] [template-name]` command to create new projects with templates.
```
Comment (rozenmd) [inline templates/README.md:15]: Isn't it `@2` now?

File: templates/pages-example-forum-app/tsconfig.json [hunk 3]
```diff
@@ -1,104 +1,14 @@
...
+		"target": "esnext",
+		"lib": ["esnext"],
+		"strict": true,
+		"moduleResolution": "node",
+		"types": ["@cloudflare/workers-types"],
+		"jsx": "react-jsx",
+		"esModuleInterop": true
 	},
-	"exclude": ["functions/**/*.{js, ts}"]
+	"exclude": ["node_modules", "dist"]
```
Comment (rozenmd) [inline templates/pages-example-forum-app/tsconfig.json:13]: @cloudflare/pages should probably weigh in whether removing `"exclude": ["functions/**/*.{js, ts}"]` is a good idea

File: templates/worker-prospector/src/functions.ts [hunk 4]
```diff
@@ -1,5 +1,4 @@
 import { fromXML } from "from-xml";
-import Puppeteer from "@cloudflare/puppeteer";
```
Comment (rozenmd) [inline templates/worker-prospector/src/functions.ts:2]: Unused?

File: templates/README.md (hunk 2 above)
Comment (admah) [reply templates/README.md:15]: @rozenmd `@2` will not currently work as expected to create templates this way. As a workaround, users wanting to copy/paste as outlined here should use v1 until c3 supports this.

**Review by petebacondarwin — APPROVED** [review]
LGTM - I didn't try any of the templates out for real though...

File: templates/worker-prospector/src/functions.ts (hunk 4 above)
Comment (JacobMGEvans) [reply templates/worker-prospector/src/functions.ts:2]: Correct

File: templates/pages-example-forum-app/tsconfig.json (hunk 3 above)
Comment (JacobMGEvans) [reply templates/pages-example-forum-app/tsconfig.json]: I'm gonna put it back for now, realistically, the TSConfig and template setup is not complex enough to necessitate a nested `tsconfig.json` in the `/functions` the root config can very easily be tuned for the whole template. 

I will do that in another PR and have Pages looped in as well, thanks for the catch Max!

**Pull request merged**